src/cpp/dds/domain/DomainParticipant.cpp

```diff
@@ -133,17 +133,25 @@ DomainParticipant& DomainParticipant::default_publisher_qos(
     return *this;
 }
 
-//dds::sub::qos::SubscriberQos DomainParticipant::default_subscriber_qos() const
-//{
-//    return this->delegate()->get_default_subscriber_qos();
-//}
+dds::sub::qos::SubscriberQos DomainParticipant::default_subscriber_qos() const
+{
+    return this->delegate()->get_default_subscriber_qos();
+}
 
-//DomainParticipant& DomainParticipant::default_subscriber_qos(
-//        const ::dds::sub::qos::SubscriberQos& qos)
-//{
-//    this->delegate()->set_default_subscriber_qos(qos);
-//    return *this;
-//}
+DomainParticipant& DomainParticipant::default_subscriber_qos(
+        const ::dds::sub::qos::SubscriberQos& qos)
+{
+    ReturnCode_t result = delegate()->set_default_subscriber_qos(qos);
+    if (result == ReturnCode_t::RETCODE_INCONSISTENT_POLICY)
+    {
+        throw dds::core::InconsistentPolicyError("Inconsistent Qos");
+    }
+    if (result == ReturnCode_t::RETCODE_UNSUPPORTED)
+    {
+        throw dds::core::UnsupportedError("Unsupported Qos");
+    }
```

Comment:
This check is not necessary as never returns UNSUPPORTED. Also, change RETCODE_INCONSISTENT_POLICY by RETCODE_IMMUTABLE_POLICY as this is the only one that can be returned apart from OK.